recipe/meta.yaml

```diff
@@ -16,36 +9,54 @@ source:
   sha256: aa5f8d32bbf6b431ac830496da7392035d6f61b4f54872f15c4bd2a9c3f536d9
 
 build:
-  number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  number: 1
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps -vv --no-build-isolation
   entry_points:
     - jupyter = jupyter_core.command:main
     - jupyter-migrate = jupyter_core.migrate:main
     - jupyter-troubleshoot = jupyter_core.troubleshoot:main
-  skip: true  # [py<38 or (python_impl == 'pypy' and (aarch64 or ppc64le))]
 
 requirements:
-  build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
   host:
     - pip
-    - python
+    - python >=3.8
     - hatchling >=1.4
+    {% if noarch_platform == "win_no_pw32" %}
+    - pypy
+    {% endif %}
   run:
     - platformdirs >=2.5
-    - python
-    - pywin32 >=300  # [win and python_impl != 'pypy']
+    - python >=3.8
     - traitlets >=5.3
+    {% if noarch_platform == "unix" %}
+    - __{{ noarch_platform }}
+    {% elif noarch_platform == "win" %}
+    - __win
+    - pywin32 >=300
+    {% elif noarch_platform == "win_no_pw32" %}
+    - __win
+    - pypy
+    {% endif %}
+  run_constrained:
+    {% if noarch_platform == "win" %}
+    - pypy <0
+    {% endif %}
+
+{% set skip = ["test_not_on_path", "test_path_priority", "test_argv0"] %}
+
+# pypy is weird
+{% if noarch_platform == "win_no_pw32" %}
+{% set skip = skip + ["migrate"] %}
+{% endif %}
 
-{% set skip = ["test_not_on_path", "test_path_priority"] %}
-# linux shebang lines have a length limit longer than the placeholder test prefix
-{% set skip = skip + ["test_argv0"] %}  # [linux]
+# combine for -k
 {% set skip = '-k "not (' + (skip | join(' or ')) + ')"' %}
 
-# coverage is very slow on pypy
+{% set cov = "" %}
```

Comment:
I think we can do without the coverage tests. Enforcing test coverage for packaging tools doesn't make a lot of sense to me.

Reply:
While I'm _slightly_ more clued into the stately pace `jupyter_core` dev than some of the other things I maintain, I've found that sudden, rapid drops in coverage are the best way to find e.g. new optional dependencies. I've been generally trying to keep them on important packages, where possible.